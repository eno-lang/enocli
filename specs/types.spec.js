const path = require('path');

const analyze = require('../lib/analyze.js');

const DOCUMENT_PATH = path.join(__dirname, 'fixtures/types.eno');

describe('analyze', () => {
  it('analyzes possible value types by inference', () => {
    const astDocument = analyze(DOCUMENT_PATH);

    expect(astDocument).toMatchSnapshot();
  });
});
