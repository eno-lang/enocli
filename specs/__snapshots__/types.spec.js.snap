// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`analyze analyzes possible value types by inference 1`] = `
Object {
  "elements": Array [
    Object {
      "internalKey": "boolean",
      "key": "boolean",
      "parent": [Circular],
      "required": true,
      "type": "field",
      "valueRequired": true,
      "valueTypes": Array [
        Object {
          "loader": [Function],
          "name": "boolean",
          "placeholder": "true (e.g.)",
        },
        Object {
          "loader": [Function],
          "name": "json",
          "placeholder": "{ \\"key\\": \\"value\\" } (e.g.)",
        },
        Object {
          "loader": [Function],
          "name": "string",
          "placeholder": "Lorem Ipsum (e.g.)",
        },
      ],
    },
    Object {
      "internalKey": "color",
      "key": "color",
      "parent": [Circular],
      "required": true,
      "type": "field",
      "valueRequired": true,
      "valueTypes": Array [
        Object {
          "loader": [Function],
          "name": "color",
          "placeholder": "#fff (e.g.)",
        },
        Object {
          "loader": [Function],
          "name": "string",
          "placeholder": "Lorem Ipsum (e.g.)",
        },
      ],
    },
    Object {
      "internalKey": "date",
      "key": "date",
      "parent": [Circular],
      "required": true,
      "type": "field",
      "valueRequired": true,
      "valueTypes": Array [
        Object {
          "loader": [Function],
          "name": "date",
          "placeholder": "2020-01-01 (e.g.)",
        },
        Object {
          "loader": [Function],
          "name": "datetime",
          "placeholder": "1997-07-07T13:15:30Z (e.g.)",
        },
        Object {
          "loader": [Function],
          "name": "string",
          "placeholder": "Lorem Ipsum (e.g.)",
        },
      ],
    },
    Object {
      "internalKey": "datetime",
      "key": "datetime",
      "parent": [Circular],
      "required": true,
      "type": "field",
      "valueRequired": true,
      "valueTypes": Array [
        Object {
          "loader": [Function],
          "name": "datetime",
          "placeholder": "1997-07-07T13:15:30Z (e.g.)",
        },
        Object {
          "loader": [Function],
          "name": "string",
          "placeholder": "Lorem Ipsum (e.g.)",
        },
      ],
    },
    Object {
      "internalKey": "email",
      "key": "email",
      "parent": [Circular],
      "required": true,
      "type": "field",
      "valueRequired": true,
      "valueTypes": Array [
        Object {
          "loader": [Function],
          "name": "email",
          "placeholder": "jane.doe@eno-lang.org (e.g.)",
        },
        Object {
          "loader": [Function],
          "name": "string",
          "placeholder": "Lorem Ipsum (e.g.)",
        },
      ],
    },
    Object {
      "internalKey": "float",
      "key": "float",
      "parent": [Circular],
      "required": true,
      "type": "field",
      "valueRequired": true,
      "valueTypes": Array [
        Object {
          "loader": [Function],
          "name": "float",
          "placeholder": "4.83 (e.g.)",
        },
        Object {
          "loader": [Function],
          "name": "json",
          "placeholder": "{ \\"key\\": \\"value\\" } (e.g.)",
        },
        Object {
          "loader": [Function],
          "name": "string",
          "placeholder": "Lorem Ipsum (e.g.)",
        },
      ],
    },
    Object {
      "internalKey": "integer",
      "key": "integer",
      "parent": [Circular],
      "required": true,
      "type": "field",
      "valueRequired": true,
      "valueTypes": Array [
        Object {
          "loader": [Function],
          "name": "integer",
          "placeholder": "14 (e.g.)",
        },
        Object {
          "loader": [Function],
          "name": "float",
          "placeholder": "4.83 (e.g.)",
        },
        Object {
          "loader": [Function],
          "name": "json",
          "placeholder": "{ \\"key\\": \\"value\\" } (e.g.)",
        },
        Object {
          "loader": [Function],
          "name": "string",
          "placeholder": "Lorem Ipsum (e.g.)",
        },
      ],
    },
    Object {
      "internalKey": "ipv4",
      "key": "ipv4",
      "parent": [Circular],
      "required": true,
      "type": "field",
      "valueRequired": true,
      "valueTypes": Array [
        Object {
          "loader": [Function],
          "name": "ipv4",
          "placeholder": "192.168.62.27 (e.g.)",
        },
        Object {
          "loader": [Function],
          "name": "string",
          "placeholder": "Lorem Ipsum (e.g.)",
        },
      ],
    },
    Object {
      "internalKey": "lat_lng",
      "key": "lat_lng",
      "parent": [Circular],
      "required": true,
      "type": "field",
      "valueRequired": true,
      "valueTypes": Array [
        Object {
          "loader": [Function],
          "name": "lat_lng",
          "placeholder": "48.205870, 16.413690 (e.g.)",
        },
        Object {
          "loader": [Function],
          "name": "string",
          "placeholder": "Lorem Ipsum (e.g.)",
        },
      ],
    },
    Object {
      "internalKey": "json",
      "key": "json",
      "parent": [Circular],
      "required": true,
      "type": "field",
      "valueRequired": true,
      "valueTypes": Array [
        Object {
          "loader": [Function],
          "name": "json",
          "placeholder": "{ \\"key\\": \\"value\\" } (e.g.)",
        },
        Object {
          "loader": [Function],
          "name": "string",
          "placeholder": "Lorem Ipsum (e.g.)",
        },
      ],
    },
    Object {
      "internalKey": "url",
      "key": "url",
      "parent": [Circular],
      "required": true,
      "type": "field",
      "valueRequired": true,
      "valueTypes": Array [
        Object {
          "loader": [Function],
          "name": "url",
          "placeholder": "https://eno-lang.org (e.g.)",
        },
        Object {
          "loader": [Function],
          "name": "string",
          "placeholder": "Lorem Ipsum (e.g.)",
        },
      ],
    },
    Object {
      "internalKey": "float_list",
      "key": "float_list",
      "parent": [Circular],
      "required": true,
      "type": "list",
      "valueTypes": Array [
        Object {
          "loader": [Function],
          "name": "float",
          "placeholder": "4.83 (e.g.)",
        },
        Object {
          "loader": [Function],
          "name": "json",
          "placeholder": "{ \\"key\\": \\"value\\" } (e.g.)",
        },
        Object {
          "loader": [Function],
          "name": "string",
          "placeholder": "Lorem Ipsum (e.g.)",
        },
      ],
      "valuesRequired": true,
    },
    Object {
      "internalKey": "datetime_list",
      "key": "datetime_list",
      "parent": [Circular],
      "required": true,
      "type": "list",
      "valueTypes": Array [
        Object {
          "loader": [Function],
          "name": "datetime",
          "placeholder": "1997-07-07T13:15:30Z (e.g.)",
        },
        Object {
          "loader": [Function],
          "name": "string",
          "placeholder": "Lorem Ipsum (e.g.)",
        },
      ],
      "valuesRequired": true,
    },
    Object {
      "internalKey": "json_list",
      "key": "json_list",
      "parent": [Circular],
      "required": true,
      "type": "list",
      "valueTypes": Array [
        Object {
          "loader": [Function],
          "name": "json",
          "placeholder": "{ \\"key\\": \\"value\\" } (e.g.)",
        },
        Object {
          "loader": [Function],
          "name": "string",
          "placeholder": "Lorem Ipsum (e.g.)",
        },
      ],
      "valuesRequired": true,
    },
  ],
  "internalKey": "document",
  "required": true,
  "type": "section",
}
`;
